frontends/web/src/routes/account/walletconnect/components/session-card/session-card.tsx

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright 2023 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CoreTypes } from '@walletconnect/types';
+import { Button } from '../../../../../components/forms';
+import { useTranslation } from 'react-i18next';
+import { useLoad } from '../../../../../hooks/api';
+import { getETHAddressByAccountCode } from '../../../../../api/account';
+import { truncateAddress } from '../../utils';
+import styles from './session-card.module.css';
+
+type TTextDataProps = {
+    accountName: string;
+    receiveAddress: string;
+    dAppName: string;
+    dAppUrl: string;
+    iconUrl: string;
+}
+
+type TWCSessionCardProps = {
+  metadata: CoreTypes.Metadata;
+  onDisconnect: () => void;
+  receiveAddress: string;
+}
+
+const TextData = ({ accountName, receiveAddress, dAppName, dAppUrl, iconUrl }: TTextDataProps) => {
+  return <div className={styles.textDataContainer}>
+
+    <div className={styles.accountNameAndWalletContainer}>
+      <p className={styles.accountName}>{accountName}</p>
+      <p className={`${styles.receiveAddress} hide-on-small`}>{receiveAddress}</p>
+    </div>
+
+    <p className={`${styles.receiveAddress} show-on-small`}>{receiveAddress}</p>
+
+    <div className={styles.dAppMetadataAndIconContainer}>
+      <div className={styles.dAppNameAndUrlContainer}>
+        <p>{dAppName}</p>
+        <p className={styles.dappUrl}>{dAppUrl}</p>
+      </div>
+      <img className="hide-on-small" src={iconUrl} alt="dApp icon" />
+    </div>
+
+  </div>;
+};
+
+
+export const WCSessionCard = ({ metadata, receiveAddress, onDisconnect }: TWCSessionCardProps) => {
+  const { t } = useTranslation();
+  const { name, url, icons } = metadata;
+  const accountDetail = useLoad(() => getETHAddressByAccountCode(receiveAddress), []);
+  const truncatedAddress = truncateAddress(receiveAddress);
+  const accountName = accountDetail && accountDetail.name ? accountDetail.name : '';
+
+  return (
+    <div className={styles.container}>
+      <TextData
+        accountName={accountName}
+        receiveAddress={truncatedAddress}
+        dAppName={name}
+        dAppUrl={url}
+        iconUrl={icons[0]}
+      />
```

Comment:
These `icons[0]` still confuse me a bit, could you check if there is ever more than 1 icon?

also is that absolute URL pointing to random urls? We should check if this image can load in Qt app, as I think we might block arbitrary domains.

If they all come from the same domain we could white list the domain. If they come from random domains we should look into other solutions. I just make a note here so we don't forget, doesn't have to be in this PR, but would be nice to find out.

Reply:
Well, we can only know the data structure because the icons are coming from the dApp (probably theres even a possibility, albeit small, that theres no icon??  but anyway i expect "production" dApps to have it).

Looking at the [documentation](https://docs.walletconnect.com/2.0/specs/clients/core/pairing/data-structures), it **will always be an array**. So it's safe to say we can try to get `icons[0]` at all times. To be extra careful we can even have a placeholder icon and use it if `icons[0]` is undef. Feel free to suggest! :)

>also is that absolute URL pointing to random urls?

Good question. Yes they're pointing to different origins. Regarding our blocking/whitelist I'm unsure how we can tackle this.. 🤔

Reply:
> These `icons[0]` still confuse me a bit, could you check if there is ever more than 1 icon?
> 
> also is that absolute URL pointing to random urls? We should check if this image can load in Qt app, as I think we might block arbitrary domains.
> 
> If they all come from the same domain we could white list the domain. If they come from random domains we should look into other solutions. I just make a note here so we don't forget, doesn't have to be in this PR, but would be nice to find out.

The URLs can be anything since they are afaik hosted on the Dapp's domain. I believe the last time we discussed this, we said if there was an issue with Qt, we could blanket whitelist all requests coming from the /wallet-connect route in the BBApp, or whatever the final route will end up where we fetch the icons